src/lib.rs

```diff
@@ -885,51 +925,747 @@ impl GooseAttack {
+                    DEFAULT_PORT.to_string().parse().unwrap()
+                };
+            }
+        } else {
+            if !self.configuration.manager_host.is_empty() {
+                return Err(GooseError::InvalidOption {
+                    option: "--manager-host".to_string(),
+                    value: self.configuration.manager_host.clone(),
+                    detail:
+                        "The --manager-host option must be set together with the --worker flag."
+                            .to_string(),
+                });
+            }
+
+            if self.configuration.manager_port != 0 {
+                return Err(GooseError::InvalidOption {
+                    option: "--manager-port".to_string(),
+                    value: self.configuration.manager_port.to_string(),
+                    detail:
+                        "The --manager-port option must be set together with the --worker flag."
+                            .to_string(),
+                });
+            }
+        }
+
+        Ok(())
+    }
+
+    // Configure how many Goose Users to hatch.
+    fn set_users(&mut self) -> Result<(), GooseError> {
+        // Use --users if set.
+        self.users = if let Some(u) = self.configuration.users {
+            if self.attack_mode == GooseMode::Worker {
+                return Err(GooseError::InvalidOption {
+                    option: "--users".to_string(),
+                    value: self.users.to_string(),
+                    detail: "The --users option can not be set together with the --worker flag."
+                        .to_string(),
+                });
+            }
+            u
+        // Otherwise, use default if set, but not on Worker.
+        } else if let Some(u) = self.defaults.users {
+            if self.attack_mode == GooseMode::Worker {
+                0
+            } else {
+                u
+            }
+        // Otherwise use number of CPUs.
+        } else {
+            if self.attack_mode != GooseMode::Manager && self.attack_mode != GooseMode::Worker {
+                info!(
+                    "concurrent users defaulted to {} (number of CPUs)",
+                    self.number_of_cpus
+                );
+            }
+            self.number_of_cpus
+        };
+
+        // Be sure a valid users value was set.
+        if self.users == 0 && self.attack_mode != GooseMode::Worker {
+            return Err(GooseError::InvalidOption {
+                option: "--users".to_string(),
+                value: self.users.to_string(),
+                detail: "The --users option must be set to at least 1.".to_string(),
+            });
+        }
+
+        if self.attack_mode != GooseMode::Manager && self.attack_mode != GooseMode::Worker {
+            debug!("users = {}", self.users);
+        }
+
+        Ok(())
+    }
+
+    // Configure maximum run time if specified, otherwise run until canceled.
+    fn set_run_time(&mut self) -> Result<(), GooseError> {
```

Comment:
@LionsAd This one also ends up in `GooseAttack`, but likely could just overload `configuration`.

Reply:
Actually, this can not live in `configuration` as the Type changes from `&str` to `usize`. When setting `--run-time` it supports a string such as `1h30m` -- this is converted and stored as a `usize` in `GooseAttack`.